src/Workspaces/Core/Portable/Extensions/NotificationOptionExtensions.cs

```diff
@@ -1,19 +1,22 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CodeStyle
 {
     internal static class NotificationOptionExtensions
     {
         public static string ToEditorConfigString(this NotificationOption notificationOption)
         {
-            if (notificationOption == NotificationOption.Silent)
-            {
-                return nameof(NotificationOption.Silent).ToLowerInvariant();
-            }
-            else
+            return notificationOption.Severity switch
             {
-                return notificationOption.ToString().ToLowerInvariant();
-            }
+                ReportDiagnostic.Suppress => EditorConfigSeverityStrings.None,
+                ReportDiagnostic.Hidden => EditorConfigSeverityStrings.Silent,
+                ReportDiagnostic.Info => EditorConfigSeverityStrings.Suggestion,
+                ReportDiagnostic.Warn => EditorConfigSeverityStrings.Warning,
+                ReportDiagnostic.Error => EditorConfigSeverityStrings.Error,
+                _ => throw ExceptionUtilities.Unreachable
```

Comment:
There's also ExceptionUtilities.UnexpectedValue() for things like enum switches.